core/src/main/java/org/infinispan/persistence/sifs/Index.java

```diff
@@ -391,7 +415,7 @@ public CompletionStage<Void> stop() throws InterruptedException {
 
          aggregateCompletionStage.dependsOn(removeSegmentsStage);
       } finally {
-         lock.readLock().unlock();
+         lock.unlockRead(stamp);
```

Comment:
Looking now, show the `stop` have a similar change with the `.whenComplete` to unlock?

Reply:
Changing locking here won't do anything as the onComplete message will prevent any other operations from being submitted. However, I should probably double check the stop sequence code to make sure it works properly. We may need to make the LogAppender wait until the stop is complete for example. But that would be a different issue if it is one.